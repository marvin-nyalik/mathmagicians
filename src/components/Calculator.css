.container {
  width: 45vw;
  height: auto;
  margin: 0 auto;
  display: flex;
  flex-direction: column;
}

.output-result {
  width: 100%;
  height: 4rem;
  font-size: 28px;
  background: rgb(138, 136, 136);
  display: flex;
  justify-content: flex-end;
  align-items: center;
  margin-right: 20px;
  color: whitesmoke;
}

.keys-container {
  width: 100%;
  display: flex;
  justify-content: flex-start;
  flex-wrap: wrap;
}

.calc-key {
  width: 100%;
}
