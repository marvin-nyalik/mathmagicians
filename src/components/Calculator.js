import { useState } from 'react';
import Button from './Buttons';
import './Calculator.css';
import calculate from '../logic/calculate';

const Calculator = () => {
  const [result, setResult] = useState({ total: null, next: null, operation: null });
  const { total, next, operation } = result;

  const handleInput = (e) => {
    const value = e.target.innerHTML;
    setResult(calculate(result, value));
  };

  const btnValues = ['AC', '+/-', '%', '\u00F7', '7', '8', '9',
    'x', '4', '5', '6', '-', '1', '2', '3', '+', '0', '.', '='];
  return (
    <div className="container">
      <div className="output-result">{`${total || ''}${operation || ''}${next || ''}`}</div>
      <div className="keys-container">
        <div className="calc-key">
          { btnValues.map((value) => (
            <Button
              key={value}
              value={value}
              handleInput={handleInput}
            />
          )) }
        </div>
      </div>
    </div>
  );
};

export default Calculator;
