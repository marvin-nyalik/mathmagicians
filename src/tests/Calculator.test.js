import { render } from '@testing-library/react';
import Calculator from '../components/Calculator';

describe('Calculator component tests', () => {
  it('Matches the snapshot', () => {
    const { container } = render(<Calculator />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
