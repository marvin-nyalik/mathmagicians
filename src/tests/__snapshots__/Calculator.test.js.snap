// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator component tests Matches the snapshot 1`] = `
<div
  class="container"
>
  <div
    class="outputresult"
  >
    0
  </div>
  <div
    class="keyscontainer"
  >
    <div
      class="calckey"
    >
      <button
        class="keys"
        type="button"
      >
        AC
      </button>
      <button
        class="keys"
        type="button"
      >
        +/-
      </button>
      <button
        class="keys"
        type="button"
      >
        %
      </button>
      <button
        class="operator-keys"
        type="button"
      >
        รท
      </button>
      <button
        class="keys"
        type="button"
      >
        7
      </button>
      <button
        class="keys"
        type="button"
      >
        8
      </button>
      <button
        class="keys"
        type="button"
      >
        9
      </button>
      <button
        class="operator-keys"
        type="button"
      >
        x
      </button>
      <button
        class="keys"
        type="button"
      >
        4
      </button>
      <button
        class="keys"
        type="button"
      >
        5
      </button>
      <button
        class="keys"
        type="button"
      >
        6
      </button>
      <button
        class="operator-keys"
        type="button"
      >
        -
      </button>
      <button
        class="keys"
        type="button"
      >
        1
      </button>
      <button
        class="keys"
        type="button"
      >
        2
      </button>
      <button
        class="keys"
        type="button"
      >
        3
      </button>
      <button
        class="operator-keys"
        type="button"
      >
        +
      </button>
      <button
        class="zero-key"
        type="button"
      >
        0
      </button>
      <button
        class="keys"
        type="button"
      >
        .
      </button>
      <button
        class="operator-keys"
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
